docs/arch/adr-25.md

```diff
@@ -0,0 +1,154 @@
+          {"task": "T0.0.0.0"},
+          {"task": "T0.0.0.1"}
+        ]
+      },
+      {
+        "frame": 0,
+        "tool": 1,
+        "toolType": "point",
+        "x": 404.61279296875,
+        "y": 583.4398803710938,
+        "value_index": 1,
+        "details": [
+          {"task": "T0.1.1.0"},
+          {"task": "T0.1.1.1"}
+        ]
+      }
+    ]
+  },
+  {
+    "task": "T0.0.0.0",
+    "taskType": "single",
+    "value": 0
+  },
+  {
+    "task": "T0.0.0.1",
+    "taskType": "dropdown",
+    "value":[
+      {"option-1": 1},
+      {"option-2": 1},
+      {"None": 1}
+    ]
+  },
+  {
+    "task": "T0.1.1.0",
+    "taskType": "single",
+    "value": 1
+  },
+  {
+    "task": "T0.1.1.1",
+    "taskType": "dropdown",
+    "value": [
+      {"option-3": 1},
+      {"option-4": 1},
+      {"option-5": 1}
+    ]
+  }
+]}
+```
+
+The sub-task identifiers follow a convention of `TASK_KEY.ANNOTATION_INDEX.TOOL_INDEX.DETAILS_INDEX`.
+
+### Drawing sub-task UI
+
+The UI will change to adopt the design of Anti-Slavery Manuscripts (ASM) with this [generalized design](https://projects.invisionapp.com/d/main#/console/12923997/396381420/preview). It will be a pseudo-modal, but with a few notable differences from a true modal:
+
+- The initial position will be near the associated mark made
+- Interactions will be allowed with the image toolbar to allow zoom, rotate, as well as opening of the tutorial, field guide, and task help. Submission of the classification should not be allowed.
+  - If the sub-tasks are required, the modal should not be closeable until the required annotations are made or the mark is deleted if cancelled
+- The dialog can be moved and resized
+
+To support movability and resizing, we will leverage [react-rnd](https://github.com/bokuweb/react-rnd) which is the same library ASM used. Grommet's `Layer` cannot be used since it is intended for actual modal or side panel use and cannot be arbitrarily positioned or moved.
+
+## Status
+
+Proposed
+
+## Consequences
+
+Flattening the annotations array for drawing task sub-tasks is conceptually consistent with the move to using workflow steps to flatten the combo task. It is however a breaking change and this change will have to be communicated to project builders. As with other classifications from the new classifier, this can be checked by the presence of `classifier_version: 2.0` in the classification metadata. In the future, we would also like to include a link to json schema for each annotation type as recommended in [ADR 07](adr-07.md).
+
+Flattening also has added benefits for Panoptes when generating classification exports. It can parse through a flattened array to convert machine readable strings to human readable strings for each task without having to check for values in a nested `details` array and then traverse it. In the raw classification export, this also benefits researchers that want to analyze the outputted CSV directly and prefer a flatter json structure.
```

Comment:
I agree with this statement for the most part, but I want to point out that additional work would have to be done for the CSV to flatten into an expected shape.  As it is if it using the `TASK_KEY.ANNOTATION_INDEX.TOOL_INDEX.DETAILS_INDEX` key to flatten the annotation object the resulting CSV file would end up with one column for each sub-task answer given  and those columns would each have one non-null row (e.g. if a volunteer draws 50 points each with 2 sub-tasks there would be 100 new columns).  

If it was instead flattened on `TASK_KEY.TOOL_INDEX.DETAILS_INDEX` but still kept the original key for mapping back to the point it would be a more manageable shape (e.g. if a volunteer draws 50 points each with 2 sub-tasks there would be 2 new columns each with a list of 50 annotations).

Reply:
If we could have a specific example of when and why a researcher prefers a flatter structure that would be great. I just know we've been asked for that in general.